src/main/java/net/ttddyy/dsproxy/proxy/jdk/JdkJdbcProxyFactory.java

```diff
@@ -75,15 +72,29 @@ public CallableStatement createCallableStatement(CallableStatement callableState
      */
     @Override
     public ResultSet createResultSet(ResultSet resultSet, ConnectionInfo connectionInfo, ProxyConfig proxyConfig) {
+        ResultSetProxyLogicFactory factory = proxyConfig.getResultSetProxyLogicFactory();
         // when proxy logic factory is specified, create a proxy
-        if (proxyConfig.getResultSetProxyLogicFactory() != null) {
+        if (factory != null) {
             return (ResultSet) Proxy.newProxyInstance(ProxyJdbcObject.class.getClassLoader(),
                     new Class[]{ProxyJdbcObject.class, ResultSet.class},
-                    new ResultSetInvocationHandler(resultSet, connectionInfo, proxyConfig));
+                    new ResultSetInvocationHandler(factory, resultSet, connectionInfo, proxyConfig));
         } else {
             return resultSet;
         }
     }
 
+    @Override
+    public ResultSet createGeneratedKeys(Statement statement, ConnectionInfo connectionInfo, ProxyConfig proxyConfig) throws SQLException {
```

Comment:
nit: Let's change the signature `Statement` -> `ResultSet`. That way `[statement|ps].getGeneratedKeys()` happens in `~ProxyLogic` and it's more explicit.